daily-challenges/2024-09/14/tunganh222Z_1409_registerPage.java

```diff
@@ -0,0 +1,79 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Random;
+
+public class tunganh222Z_1409_registerPage {
+    WebDriver driver;
+    WebDriverWait explicitWait;
+
+    String userName = "TungAnh";
+    String email = getEmailRand();
+    @BeforeClass
+    public void beforeClass(){
+        driver = new FirefoxDriver();
+        explicitWait = new WebDriverWait(driver,Duration.ofSeconds(15));
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(15));
+        driver.manage().window().maximize();
+    }
+
+
+    @Test
+    public void TC_01_registerPage(){
+        driver.get("https://material.playwrightvn.com/");
+        driver.findElement(By.xpath("//a[contains(text(),'Register Page')]")).click();
+
+        By userName = By.xpath("//input[@id='username']");
+        By email = By.xpath("//input[@id='email']");
+        explicitWait.until(ExpectedConditions.visibilityOfElementLocated(userName));
+        driver.findElement(userName).sendKeys(this.userName);
+        driver.findElement(email).sendKeys(this.email);
+
+        driver.findElement(By.xpath("//button[@type='submit']")).click();
+
+        By informationTable = By.xpath("//tbody");
+        explicitWait.until(ExpectedConditions.visibilityOfElementLocated(informationTable));
+
+        List<WebElement> infor = driver.findElements(By.xpath("//tbody/tr/td"));
+        boolean checkUsername = false;
+        boolean checkEmail = false;
+
+        for (int i = 0 ; i < infor.size() ; i++){
```

Comment:
Đoạn code này chạy được nhưng có vẻ chưa tối ưu lắm: đối với mỗi td, sẽ check cả username và email.
Vì cấu trúc bảng luôn là cột username số 2, cột email số 3, nên hoàn toàn có thể check trực tiếp.
Code sẽ như này:

// Assert bảng có 1 row mới xuất hiện
// Assert cột số 2 của row chứa text là username vừa nhập
// Assert cột số 3 của row chứa text là email vừa nhập

Reply:
đoạn này ban đầu em cũng assert theo index nhưng em sợ sau update thay đổi cột thì em đang để theo vòng lặp để duyệt ạ, để em sửa lại assert theo index